index.js

```diff
@@ -1,7 +1,16 @@
 'use strict';
+
+function flatten(arr) {
+	return arr.reduce((previous, current) => {
+		return previous.concat(Array.isArray(current) ? flatten(current) : current);
+	}, []);
+}
+
 module.exports = function (arr) {
-	var rest = [].concat.apply([], [].slice.call(arguments, 1));
-	return arr.filter(function (el) {
-		return rest.indexOf(el) === -1;
-	});
+	const rest = new Set(
+		flatten(
+			[].slice.call(arguments, 1)
+		)
+	);
```

Comment:
We can make use of [simple-swizzle](https://github.com/Qix-/node-simple-swizzle) here. Not sure if @sindresorhus wants a dependency for something like that but it could clean up the code drastically.